crates/derive/src/stages/l1_traversal.rs

```diff
@@ -108,8 +108,19 @@ impl<F: ChainProvider + Send> OriginAdvancer for L1Traversal<F> {
         }
 
         crate::set!(ORIGIN_GAUGE, next_l1_origin.number as i64);
+
         self.block = Some(next_l1_origin);
         self.done = false;
+
+        // If the prev block is not holocene, but the next is, we need to flag this
+        // so the pipeline driver will reset the pipeline for holocene activation.
+        let prev_block_holocene =
+            self.rollup_config.is_holocene_active(self.block.unwrap().timestamp);
+        let next_block_holocene = self.rollup_config.is_holocene_active(next_l1_origin.timestamp);
+        if !prev_block_holocene && next_block_holocene {
```

Comment:
There's a bug here - we're setting `self.block` to `next_l1_origin`, then comparing it with itself.